src/slots/base.js

```diff
@@ -0,0 +1,30 @@
+import symbols from "../util/symbols.js";
+import Slots from "./slots.js";
+import SlotController from "./slot-controller.js";
+
+export const { slots } = symbols.known;
+
+export const hooks = {
+	constructed () {
+		this[slots] = SlotController.create(this);
+	},
+};
+
+export const providesStatic = {
+	defineSlots (def = this[slots] ?? this.slots) {
+		if (!(this[slots] instanceof Slots)) {
+			this[slots] = new Slots(this, def);
+		}
+		else if (this[slots] === def) {
+			// Nothing to do here
+			return;
+		}
+
+		// New slots to add
+		this[slots].add(def);
```

Comment:
```suggestion
		this[slots].define(def);
```

I think this method should be here (at least there's no `add()` method, and `define ()` does add slots).